patroni/__init__.py

```diff
@@ -145,3 +141,42 @@ def main():
         else:
             patroni.ha.while_not_sync_standby(lambda: patroni.postgresql.stop(checkpoint=False))
             patroni.dcs.delete_leader()
+
+
+def main():
+    if os.getpid() != 1:
+        return patroni_main()
+
+    pid = 0
+
+    # Looks like we are in a docker, so we will act like init
+    def sigchld_handler(signo, stack_frame):
+        try:
+            while True:
+                ret = os.waitpid(-1, os.WNOHANG)
+                if ret == (0, 0):
+                    break
+                elif ret[0] != pid:
+                    logging.info('Reaped unknow pid=%s, exit status=%s', *ret)
```

Comment:
Should be 'unknown', but actually everything except for the child Patroni pid will be considered 'unknown' (including PostgreSQL and archive/restore command), so there is no point in mentioning it (showing the pid and ret status could still be useful for debugging purposes).